control-plane/src/e3proxy.rs

```diff
@@ -55,53 +42,17 @@ impl E3Proxy {
                 }
             };
             println!("Crypto request received");
-            let e3_ip = match self.get_ip_for_e3().await {
-                Ok(Some(ip)) => ip,
-                Ok(None) => {
-                    eprintln!("No ip returned for E3");
-                    Self::shutdown_conn(connection).await;
-                    continue;
-                }
-                Err(e) => {
-                    eprintln!("Error obtaining IP for E3 — {e:?}");
-                    Self::shutdown_conn(connection).await;
-                    continue;
-                }
-            }; // TODO: cache
-            println!("IP for E3 obtained: {e3_ip}");
-            tokio::spawn(async move {
-                let e3_stream = match tokio::net::TcpStream::connect(e3_ip).await {
-                    Ok(e3_stream) => e3_stream,
-                    Err(e) => {
-                        eprintln!("Failed to connect to E3 ({e3_ip}) — {e:?}");
-                        Self::shutdown_conn(connection).await;
-                        return;
-                    }
-                };
 
+            let e3_stream = e3_client.new_conn().await.unwrap();
```

Comment:
Might be good to do the pattern match here and shut down connection on error from the existing code